### ISSUE TITLE
Check impact on supporting UUIDs as Process ID / Key

### ISSUE THREAD

**Issue by salaboy** [Activiti/Activiti#2115]
Currently the bpmn process tag has an ID property and currently this ID cannot be a UUID because of the bpmn schema which restricts this property to start with a char different from a number. 

In order to avoid this limitation we can rely on a convention such as:
ID-{UUID}

We need to check if this impacts any of the other services such as Query and Audit which are expecting a simple string to be the KEY of the process (which is used to also start process instances).

We also need to check our REST endpoint implementation, which should use the same UUID as resource ID for GET /process-definitions/{id}

**Comment by balsarori** [comment]
Seems that this restriction is part of the standard BPMN schema (see https://www.omg.org/spec/BPMN/20100501/Semantic.xsd) not Activiti specific. The ``id`` attribute only accepts a value of type xsd:ID which can't start with an number.

In APS modeller we used to overcome this issue by added an ``sid-`` prefix to all UUID elements.

**Comment by magemello** [comment]
A possible solution could be prefix the UUID with the resource type, for example:
- processes ID  : process-123-123-123-123
- connector ID  : connector-123-123-123-123
- form ID: form-123-123-123-123

So in general {resourceType}-UUID where resource type can be processes, connector, form, rule ecc...

**Comment by salaboy** [comment]
:+1 let's do that.. so we can standardise across all resources.

**Comment by erdemedeiros** [comment]
@salaboy concerning 

> We also need to check our REST endpoint implementation, which should use the same UUID as resource ID for GET /process-definitions/{id}

**Comment by salaboy** [comment]
@erdemedeiros yes.. exactly.. and we have  more resources that the one listed here:
- process instance
- process definition
- tasks
- process definition metadata

etc
